c2rust-analyze/src/main.rs

```diff
@@ -449,13 +449,62 @@ fn run(tcx: TyCtxt) {
             assert_eq!(local, l);
         }
 
+        // Find all calls to `malloc`, `calloc`, `realloc`, and `free`
+        // and track their destination locals as being c_void
+        // pointers to special-case downstream casts
+        for bb_data in mir.basic_blocks().iter() {
+            if let Some(term) = &bb_data.terminator {
+                let term: &Terminator = term;
+                if let TerminatorKind::Call {
+                    ref func,
+                    ref args,
+                    destination,
+                    ..
+                } = term.kind
+                {
+                    let func_ty = func.ty(&mir.local_decls, tcx);
+                    let mut add_c_void_ptr = |p: &Place<'tcx>| {
+                        let deref_ty = p
+                            .ty(acx.local_decls, acx.tcx())
+                            .ty
+                            .builtin_deref(true)
+                            .unwrap()
+                            .ty
+                            .kind();
+                        assert_matches!(deref_ty, TyKind::Adt(adt, _) => {
+                            assert_eq!(tcx.def_path(adt.did()).data[0].to_string(), "ffi");
+                            assert_eq!(tcx.item_name(adt.did()).as_str(), "c_void");
+                        });
+
+                        acx.c_void_ptrs.insert(*p);
+                    };
+
+                    use crate::util::Callee::*;
```

Comment:
I think it can just be `use Callee::*` with a top-level `use crate::util::Callee`.

Reply:
715944f6